### PR TITLE
Rework MyChartAPI to get multiple sites

### PR DESCRIPTION
**Pull request by Kathleen-Francis**
tl;dr - this PR consolidates/clarifies the UMassMemorialMarlborough and UMassMercantile scrapers into a single UMassMemorial scraper that returns availability for:
- UMass Memorial Marlborough
- UMass Memorial Mercantile Center
- UMMHC HealthAlliance Clinton Hospital (https://github.com/macovidvaccines-com/covid-vaccine-scrapers/issues/8)

To allow for that, I had to make changes to the MyChartAPI library to be aware of different sites in the same API response.

---

Previous MyChartAPI implementation was ignoring the possibility of the API returning data from multiple vaccination sites (`Departments` in the API response). That works if each scraper correctly filters their API request for only their own site, but that does not seem to be what happened.

Specifically, the current implementations for UMassMemorialMarlborough and UMassMercantile seem to be incorrect:
- UMassMemorialMarlborough is filtering on `"Departments":{"102001144":true,"104001144":true,"111029146":true}` ([here](https://github.com/livgust/covid-vaccine-scrapers/blob/master/site-scrapers/UMassMemorialMarlborough/index.js)). From looking at the `AllDepartments` info from the API response, it's returning data from three different sites:
  - `104001144`: UMMHC HealthAlliance Clinton Hospital Leominster Campus – Simonds Building Guild Conference Vaccine Clinic
  - `111029146`: UMass Memorial Healthcare 100 Front St Vaccine Clinic
  - `111029148`: UMMHC Marlborough Hospital 75 Felton St Vaccine Clinic
** I'm not sure why the department filter doesn't match the departments returned (102001144 vs. 111029148)
- UMassMercantile is filtering on `"Departments":{"102001144":true,"104001144":true,"111029146":true}` ([here](https://github.com/livgust/covid-vaccine-scrapers/blob/7f308159ef86cd5b5c5e46be2990d4eedd95456c/site-scrapers/UMassMercantile/index.js#L38)), the same as UMassMemorialMarlborough. It returns the same three departments, with no indication of `Mercantile` anywhere.

### CODE REVIEW

**Comment by Kathleen-Francis** [comment]
Turns out the UMass Mercantile is the same as this location (from `AllDepartments`):
```
'111029146': {
      Name: 'UMass Memorial Healthcare 100 Front St Vaccine Clinic',
      ...
    },
```
100 Front St is around the corner from (equivalent to?) 201 Commercial St ([Mercantile Center address](https://www.umassmemorialhealthcare.org/umass-memorial-health-care/patients-visitors/coronavirus-resources/covid-19-vaccination-information/covid-19-vaccination-center-the-mercantile-center)), so I will just remove the UMassMercantile scraper and update the naming for that department/site within the UMassMemorial scraper.

File: lib/MyChartAPI.js [hunk 1]
```diff
@@ -89,38 +89,74 @@ async function GetSchedule(
...
+ * Add relevant site info to the results object
+ * @param {object} results
+ * @param {object} defaultResults
+ * @param {string} siteKey
+ * @param {object} siteInfo
+ */
+function AddSiteInfo(results, defaultResults, siteKey, siteInfo) {
+    // set default availability values
+    results[siteKey] = JSON.parse(JSON.stringify(defaultResults));
+    if (Object.keys(siteInfo) == 0) return;
```
Comment (livgust) [inline lib/MyChartAPI.js:102]: are we checking for length here? IDK when Object.keys() will return 0.
Also, we should use `===`

File: lib/MyChartAPI.js [hunk 2]
```diff
@@ -138,12 +174,12 @@ async function AddFutureWeeks(
     weekCount,
     postDataCallback
 ) {
-    const results = { availability: {}, hasAvailability: false };
+    var results = {};
```
Comment (livgust) [inline lib/MyChartAPI.js:177]: We shouldn't be using `var` anywhere in this project - instead we'll use `const` or `let`. If you have a variable that needs reassignment, you'll use `let`; otherwise, `const`. Gotcha: objects can have their keys and values modified while still being `const`s (because of pointers and all that jazz).

**Review by livgust — CHANGES_REQUESTED** [review]
Love this! And thanks for throwing in some tests, too. I have 2 nitpicky comments.
Also, I have that in-progress branch boston-medical-center that would be so easy for you to pick up and finish if you want ;) (no pressure, and not in this PR obvz)

**Comment by Kathleen-Francis** [comment]
Thanks for the JS tips! Haven't worked in javascript in a few years. I can probably pick up the boston-medical-center branch in the next couple days.

File: lib/MyChartAPI.js [hunk 3]
```diff
@@ -89,38 +89,74 @@ async function GetSchedule(
...
+        const key = dayEntry.DepartmentID;
+        if (!results[key]) {
+            AddSiteInfo(results, defaultResults, key, {});
+        }
+
+        //force midnight local time zone to avoid UTC dateline issues
+        const date = new Date(`${dayEntry.DateISO}T00:00`);
+        const slots = dayEntry.Slots;
+        const slotsAvailable = slots.length || 0;
+        let midnightToday = new Date();
```
Comment (zpeyton) [inline lib/MyChartAPI.js:149]: We are switching to moment() for all date things.

File: test/MyChartAPITest.js [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+
+describe("UpdateResults", () => {
+    it("should add site info and default data to results object", () => {
+        results = {};
+        mychart.UpdateResults(results, exampleWithAvailability);
+
+        for (const [key, info] of Object.entries(results)) {
+            if (key == 10098243) {
+                // this is the only site that has availability
+                expect(info.hasAvailability).to.equal(true);
```
Comment (zpeyton) [inline test/MyChartAPITest.js:400]: Can we add a date check in here to verify the dates are right if we find appointments?

**Review by zpeyton — CHANGES_REQUESTED** [review]

**Review by zpeyton — APPROVED** [review]
Tested locally. Passes the test written and scraper correctly finds no appointments.
